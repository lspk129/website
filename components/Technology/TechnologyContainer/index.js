import styled from 'styled-components';

const TechnologyContainer = styled.div`
  text-align: center;
  display: grid;
  grid-template-columns: 375px 375px;
  grid-template-rows: 260px;
  padding-bottom: 100px;
  justify-content: center;
  align-content: center;
`;

export default TechnologyContainer;
