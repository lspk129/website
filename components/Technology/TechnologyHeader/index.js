import styled from 'styled-components';

const TechnologyHeader = styled.h2`
  text-align: center;
  padding: 60px;
`;

export default TechnologyHeader;
