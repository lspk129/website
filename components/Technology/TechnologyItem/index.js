import styled from 'styled-components';

const TehcnologyItem = styled.div`
  display: grid;
  align-items: center;
  img {
    width: 200px;
    height: 200px;
    align-self: center;
    justify-self: center;
  }
  img[alt="Next"] {
    width: 200px;
    height: 150px;
    margin: 25px 0px;
  }
`;

export default TehcnologyItem;
