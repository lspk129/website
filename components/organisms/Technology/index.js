import React from 'react';
import { Row, Col } from 'react-styled-flexboxgrid';

import Card from '../../molecules/Card';
import Text from '../../atoms/Text';
import Division from '../../atoms/Division';
import DivisionHover from '../../atoms/DivisionHover';

const Technology = () => (
  <Division pb={3}>
    <Division py={3}>
      <Text tag={'h2'}>
        We use cutting-edge technologies
      </Text>
    </Division>
    <Row center={'xs'}>
      <Col md={3} sm={4} xs={12}>
        <DivisionHover scale={1.1}>
          <Card
            image={<img src="/static/React_logo.png" alt="React" />}
            label={'React'}
          />
        </DivisionHover>
      </Col>
      <Col md={3} sm={4} xs={12}>
        <DivisionHover scale={1.1}>
          <Card
            image={<img src="/static/Meteor_logo.png" alt="Meteor" />}
            label={'Meteor'}
          />
        </DivisionHover>
      </Col>
      <Col md={3} sm={4} xs={12}>
        <DivisionHover scale={1.1}>
          <Card
            image={<img src="/static/Nextjs_logo.png" alt="Next.js" />}
            label={'Next.js'}
          />
        </DivisionHover>
      </Col>
    </Row>
  </Division>
);

export default Technology;
