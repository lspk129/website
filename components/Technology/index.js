import React from 'react';

import TechnologyHeader from './TechnologyHeader';
import TechnologyContainer from './TechnologyContainer';
import TechnologyItem from './TechnologyItem';

const Technology = () => (
  <div>
    <TechnologyHeader>
      We use cutting-edge technologies
    </TechnologyHeader>
    <TechnologyContainer>
      <TechnologyItem>
        <img src="../../static/react_logo.png" alt="React" />
        <h3>React Js</h3>
      </TechnologyItem>
      <TechnologyItem>
        <img src="../../static/meteor_logo.png" alt="Meteor" />
        <h3>Meteor Js</h3>
      </TechnologyItem>
      <TechnologyItem>
        <img src="../../static/next_logo.png" alt="Next" />
        <h3>Next.js</h3>
      </TechnologyItem>
    </TechnologyContainer>
  </div>
);

export default Technology;
