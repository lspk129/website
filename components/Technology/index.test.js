import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

import Technology from './index';
import TechnologyHeader from './TechnologyHeader';
import TechnologyContainer from './TechnologyContainer';
import TechnologyItem from './TechnologyItem';

describe('Technology ', () => {
  const shallowWrapper = shallow(<Technology />);

  it('renders correctly', () => {
    const wrapper = renderer.create(<Technology />);
    const tree = wrapper.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders TechnologyHeader', () => {
    expect(shallowWrapper).toBePresent();
    expect(shallowWrapper.find(TechnologyHeader)).toBePresent();
  });

  it('renders h2 tag with content "We use cutting-edge technologies"', () => {
    const wrapper = mount(<Technology />);
    expect(wrapper.find(TechnologyHeader)).toHaveText('We use cutting-edge technologies');
  });

  it('renders TechnologyContainer', () => {
    expect(shallowWrapper.find(TechnologyContainer)).toBePresent();
  });

  it('renders TechnologyItem', () => {
    expect(shallowWrapper.find(TechnologyItem)).toBePresent();
  });

  it('renders img tag to the technologies section', () => {
    expect(shallowWrapper.find('img').at(0)).toHaveProp('src', '../../static/react_logo.png');
  });

  it('renders h3 tag with content "React Js"', () => {
    expect(shallowWrapper.find('h3').at(0)).toHaveText('React Js');
  });

  it('renders img tag to the technologies section', () => {
    expect(shallowWrapper.find('img').at(1)).toHaveProp('src', '../../static/meteor_logo.png');
  });

  it('renders h3 tag with content "Meteor Js"', () => {
    expect(shallowWrapper.find('h3').at(1)).toHaveText('Meteor Js');
  });

  it('renders img tag to the technologies section', () => {
    expect(shallowWrapper.find('img').at(2)).toHaveProp('src', '../../static/next_logo.png');
  });

  it('renders h3 tag with content "Next.js"', () => {
    expect(shallowWrapper.find('h3').at(2)).toHaveText('Next.js');
  });
});
