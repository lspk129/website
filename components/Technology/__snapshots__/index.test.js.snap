// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Technology  renders correctly 1`] = `
<div>
  <h2
    className="sc-bdVaJa hbgqAY"
  >
    We use cutting-edge technologies
  </h2>
  <div
    className="sc-bwzfXH hKwqyr"
  >
    <div
      className="sc-htpNat gPfKWp"
    >
      <img
        alt="React"
        src="../../static/react_logo.png"
      />
      <h3>
        React Js
      </h3>
    </div>
    <div
      className="sc-htpNat gPfKWp"
    >
      <img
        alt="Meteor"
        src="../../static/meteor_logo.png"
      />
      <h3>
        Meteor Js
      </h3>
    </div>
    <div
      className="sc-htpNat gPfKWp"
    >
      <img
        alt="Next"
        src="../../static/next_logo.png"
      />
      <h3>
        Next.js
      </h3>
    </div>
  </div>
</div>
`;
